docs/architecture/adr/ADR037-per-namespace-gateways.md

```diff
@@ -0,0 +1,28 @@
+# ADR037: Per namespace istio gateways
+
+## Status
+
+Pending
+
+## Context
+
+Different tenant applications have differing ingress and egress requirements. Examples include:
+
+* mutual TLS
+* egress controls (e.g. limiting what can egress from a namespace / cluster)
+* certificate operations (renewal etc.)
+
+At present, a single ingress gateway and egress gateway are defined by the GSP. In the case of the ingress gateway, all ingress traffic is routed through this which creates a range of difficulties:
+
+* secrets relating to TLS certs (mutual or otherwise) have to be stored in `istio-system`
```

Comment:
```suggestion
* secrets relating to TLS certs (mutual or otherwise) have to be stored in `istio-system`
  * this means that, for example, tenants who wish to implement mTLS using tenant-provided certs need to have a way of getting those certs into a Secret in `istio-system`.  It also means that rotating those certificates will require tenants to be able to make changes to `istio-system`.  This is problematic because `istio-system` is a privileged area and tenants should not normally need access to it.
```

Reply:
I know we have experienced this example, so it's worth capturing the specific use case.

Reply:
I _think_ I've addressed this.